table/plain/plain_table_reader.cc

```diff
@@ -131,10 +131,10 @@ Status PlainTableReader::Open(
   auto s = ReadTableProperties(file.get(), file_size, kPlainTableMagicNumber,
                                ioptions, &props_ptr,
                                true /* compression_type_missing */);
-  std::shared_ptr<TableProperties> props(props_ptr);
   if (!s.ok()) {
     return s;
   }
+  std::shared_ptr<TableProperties> props(props_ptr);
```

Comment:
Maybe I'm wrong, but in this case props_ptr fails to be destroyed in the not-OK case, so it will be a memory leak. Did I miss anything?